quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/infinispan/jgroups/impl/CertificateEntity.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2025 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.quarkus.runtime.storage.infinispan.jgroups.impl;
+
+import java.security.KeyPair;
+import java.security.cert.X509Certificate;
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.keycloak.common.util.PemUtils;
+
+/**
+ * JPA entity to store the {@link X509Certificate} and {@link KeyPair}.
+ */
+@SuppressWarnings("unused")
+public class CertificateEntity {
```

Comment:
Thi is rather a generic comment. I would assume most of the logic around storing the key material etc. would be Quarkus agnostic. I would rather keep it outside of the `quarkus` module which is mostly meant as an integration layer between Quarkus and KC. But it's not a blocker.

Reply:
@vmuzikar are you suggesting creating a new Maven module?